### ISSUE TITLE
F-53 · A daily-bar backtest cannot exercise the daily loss limit at all

### ISSUE THREAD

**Issue by imitusov** [imitusov/bazarabot#53]
**Severity:** Limit · **Area:** research fidelity
**Effect:** the daily loss limit — the control that stops the bot after a bad day — is structurally unreachable in a backtest. Every reported run implicitly assumes it never fires, and no backtest can show whether it would have.

Found while writing #49's guard: a behavioural test that tried to trip the limit passed with a live defect still present, because the limit cannot trip.

## Cause

`backtest.run` drives one `trading_cycle` per bar, and daily bars mean **one cycle per Moscow date**. Step 4 does both halves in that single cycle:

```
await _write_opening_snapshot(moment, open_count)
loss = await daily_loss_pct(moment)
```

The opening baseline is written and then measured against **in the same instant**, so the measured intra-day loss is always ~0. `halt(HaltReason.DAILY_LOSS_LIMIT, ...)` is unreachable, and so is everything downstream of it — the halt persisting, entries stopping, `/resume` clearing it.

## Why this matters beyond the test

A backtest's job is to say what the live system would have done. On any run where the strategy would have lost more than `DAILY_LOSS_LIMIT_PCT` in a day, live would have **stopped trading for the rest of that day** and the backtest keeps going. The simulated result is therefore not conservative — it is optimistic in exactly the scenario the limit exists for, and the divergence is invisible in the output.

It is the same shape as #12's original stop-on-the-close defect: the simulation is systematically wrong in the direction that flatters.

## Fix — needs a decision

1. **More than one cycle per bar.** Drive N cycles across a bar using open/high/low/close as successive marks, so an intra-bar drawdown is visible to `pnl.bot_equity` while the day's opening snapshot stays fixed. Cheap, approximate, and it makes the limit reachable.
2. **Intraday bars.** The honest version, and the same answer #13 wants. Costs a data pipeline the project does not have.
3. **Accept and document.** State in the `BacktestResult` contract that the daily loss limit is not modelled, so a reader knows the number excludes it.

(1) is probably the right first step: it makes the control testable at all, and it composes with (2) later. (3) alone is not enough — a result that silently omits a risk control is the kind of number that gets trusted.

## Verification
- A run whose equity falls more than `DAILY_LOSS_LIMIT_PCT` within one bar halts, and opens no further position that day.
- A run that recovers the next day resumes, since the halt is per-day.
- The behavioural version of #49's alert guard becomes possible, and passes.

## Related
#12 (the rebuild), #13 (intraday evaluation), #49 (where this surfaced), #9 (the limit's own definition).

**Comment by imitusov** [comment]
Closed by spec v1.49/v1.50 and two commit pairs on main (`be62257`), taking option 1 — multiple cycles per bar.

## What changed

`backtest.run` now drives **four cycles per bar** at the open, low, high and close. `SimulatedExchange.advance(moment, phase)` reports that phase of each instrument's current bar as its last price, and standing stops are checked **once per bar** on first entry, so four marks are not four chances to fire.

Order is **open, low, high, close** — the drawdown before the recovery, the same pessimism as the stop-beats-target tie-break, and for the same reason: a daily bar cannot say which came first.

The last mark sits five minutes before the session end, inside `in_closing_window`'s fifteen.

## It fixed a second unreachable control I found while measuring

`MAX_AGE` was **also** structurally unreachable — `lifecycle.exits` requires `in_closing_window(now)` and the single cycle sat at the session start. Measured before: `max_holding_days=1` over twenty flat bars with stop and target 50% away produced **zero exits**. After: 9.

And a third thing that was never filed: `get_last_price` returned the bar's **close**, so a `LOCAL` position's stop was checked against the close only — the exact optimism #12 removed for exchange stops, still present on the path where the bot owns the stop. Walking the low fixes it. The test for it needed the double to be able to refuse a stop order, so `reject_stops` was added — which is also the only way a backtest can reach rule 23's degrade path at all.

## A design correction along the way

`advance` first took a `mark: Decimal`. Wrong: the marks are per instrument and a backtest runs the whole watchlist, so one ticker's low would have been reported as every ticker's last price. It takes a `Phase` and the exchange resolves it per instrument (spec v1.50). Caught while writing the driver, which is the second time this session that writing the consumer exposed a flaw in the producer.

## The results move, a lot

Same 180 bars × 4 tickers, `ma_crossover`, shipped `MAX_HOLDING_DAYS=3`:

| | before | after |
|---|---|---|
| exits | 10 STOP_LOSS, 2 TAKE_PROFIT | **13 MAX_AGE** |
| realised | −7746 | +3580 |
| drawdown | 8.23% | 1.99% |

Every position now exits on age. Verified that is the fix and not a new defect by loosening the cap: at `max_holding_days=8` all three triggers reappear (9 MAX_AGE, 2 STOP_LOSS, 1 TAKE_PROFIT), so the stop machinery is intact — positions were simply exiting on age before a 5% stop was touched.

**Which means the shipped `MAX_HOLDING_DAYS=3` dominates the other two exits**, and every backtest run before today said the opposite because it could not fire the trigger that actually governs. That is a strategy-parameter finding, not a code one, and it is exactly the sort of thing this issue existed to make visible.

## One case still not reachable

A backtest that ends mid-position logs `ExitFailed: exit outcome unknown` on the last bar, because a sell has no next open to fill at. That is correct — history ran out — and matches the fill rule, but it means the final bar's exits are never booked.

## Related
Also see #13: live polls intra-day and can act within a bar; four marks narrow that gap but do not close it.
